department-of-veterans-affairs/Virtual-Regional-Office
# Houli/python version update

# Impetus

The version of poetry that is a part of the Circle CI python Docker image `cimg/python:3.8` produces error `"[TypeError] unhashable type: 'VersionUnion'"` (This error happens in our repo, due to some code in PR #21.)

I think this error happens during our `$(POETRY) install` command.

Rather than submitting a GitHub issue on the Poetry docker image repo to ask them to upgrade the poetry version used by Docker image `cimg/python:3.8`, in this PR I decided instead to just upgrade our entire project from Python 3.8.6 to 3.9.7

Why 3.9.7 specifically?
- Lambda now supports Python 3.9. That's the newest it supports.
- The newest 3.9.x Python version is 3.9.7.

# To Test

Note: If your AWS SAM CLI is too old, AWS SAM CLI will fail when doing `sam build`, saying that AWS doesn't support Python 3.9.

The fix?

Upgrade your AWS SAM CLI.

I ran into the problem when using AWS SAM CLI 1.27.2. When I upgraded, it took me to AWS SAM CLI 1.33.0.

After this, AWS SAM CLI no longer complained about Python 3.9 when I did a `sam build`.

Other than that, everything should just work...

`make build.and.pytest`

`make deploy.stack`
